pkgs/by-name/up/upscaler/package.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  fetchFromGitLab,
+  meson,
+  ninja,
+  gitUpdater,
+  desktop-file-utils,
+  appstream,
+  blueprint-compiler,
+  python3Packages,
+  pkg-config,
+  libadwaita,
+  wrapGAppsHook4,
+  upscayl-ncnn,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "upscaler";
+  version = "1.4.0";
+
+  pyproject = false; # meson instead of pyproject
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Upscaler";
+    rev = version;
+    hash = "sha256-Dy8tykIbK5o0XulurG+TxORZZSxfRe5Kjh6aGpsh+0Y=";
+  };
+
+  passthru.updateScript = gitUpdater { };
+
+  postPatch = ''
+    substituteInPlace upscaler/window.py \
+      --replace-fail '"upscayl-bin",' '"${lib.getExe upscayl-ncnn}",'
+  '';
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    meson
+    ninja
+    desktop-file-utils
+    appstream
+    blueprint-compiler
+    pkg-config
+  ];
+
+  dependencies = with python3Packages; [
+    pygobject3
+    pillow
+    vulkan
+  ];
+
+  buildInputs = [
+    libadwaita
+    upscayl-ncnn
+  ];
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = {
+    description = "Upscale and enhance images";
+    homepage = "https://tesk.page/upscaler";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ grimmauld ];
```

```suggestion
    maintainers = with lib.maintainers; [ grimmauld getchoo ];
```
And feel free to add me here as well :)

---

(add me as well)

---

On the deps too, or just upscaler?

---

I'd be fine with the deps too